metasynth/__main__.py

```diff
@@ -80,13 +108,58 @@ def main():
     elif args.output.suffix == ".xlsx":
         data_frame.write_excel(args.output)
     elif args.output.suffix == ".pkl":
-        with open(args.output, "wb") as pkl_file:
+        with args.output.open("wb") as pkl_file:
             pickle.dump(data_frame, file=pkl_file)
     else:
         parser.error(
-            f"Unsupported output file format ({args.output.suffix})." +
-            "Use .csv, .feather, .parquet, .pkl, or .xlsx."
+            f"Unsupported output file format ({args.output.suffix})."
+            "Use .csv, .feather, .parquet, .pkl, or .xlsx.",
+        )
+
+
+def schema() -> None:
+    """Program to generate json schema from dist providers."""
+    parser = argparse.ArgumentParser(
+        prog="metasynth schema",
+        description="Create Generative Metadata Format schema and print to console.",
+    )
+
+    parser.add_argument(
+        "plugins",
+        help="Plugins to include in the generated schema (default builtin)",
+        nargs="*",
+        default="builtin"
+    )
+
+    parser.add_argument(
+        "-l", "--list",
+        help="display available plugins and quit",
+        action="store_true",
+    )
+
+    # parse the args without the subcommand
+    args, _ = parser.parse_known_args()
+
+    # deduplicated list of plugins for schema
+    plugins_avail = {entry.name for entry in entry_points(group="metasynth.distribution_provider")}
+
+    if args.list:
```

Comment:
I think for now it's okay to make "builtin" special, but I think if other packages are coming up, we might want to not make it special.